into/backends/ssh.py

```diff
@@ -0,0 +1,107 @@
+from __future__ import absolute_import, division, print_function
+
+import paramiko
+from contextlib import contextmanager
+from toolz import keyfilter, memoize, take
+from datashape import discover
+import re
+
+from ..utils import keywords, tmpfile
+from ..resource import resource
+
+
+class _SSH(object):
+    """ Parent class for data accessed through ``ssh``
+
+    See ``paramiko.SSHClient.connect`` for authentication keyword arguments
+
+    Examples
+    --------
+
+    >>> from into import SSH, CSV
+    >>> s = SSH(CSV)('/path/to/file.csv', hostname='hostname', username='alice')
+
+    Normally create through resource uris
+
+    >>> data = resource('ssh://alice@host:/path/to/file.csv', password='pass')
+    >>> data.path
+    '/path/to/file.csv'
+    >>> data.auth['hostname']
+    'host'
+    """
+    def __init__(self, *args, **kwargs):
+        self.auth = keyfilter(keywords(paramiko.SSHClient.connect).__contains__,
+                              kwargs)
+        self.subtype.__init__(self, *args, **kwargs)
+
+    @memoize
+    def connect(self):
+        ssh = paramiko.SSHClient()
+        ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+        ssh.connect(**self.auth)
+        return ssh
+
+    def lines(self):
+        ssh = self.connect()
+        sftp = ssh.open_sftp()
+        return sftp.file(self.path, 'r')
+
```

Comment:
Actually, I intended to write:
`<username:xxxxxxxx@hostname:port>`

Filename is a good point -- perhaps:
`<username:xxxxxxxx@hostname:port -- basename>`